src/OnlineVotingS.Application/Services/ImplService/CampaignService.cs

```diff
@@ -12,71 +13,176 @@
 
 namespace OnlineVotingS.Application.Services.ImplService;
 
-    public class CampaignService : ICampaignService
+public class CampaignService : ICampaignService
+{
+    private readonly ICampaignRepository _campaignRepository;
+    private readonly IMapper _mapper;
+    private readonly ILogger<CampaignService> _logger;
+
+    public CampaignService(ICampaignRepository campaignRepository, IMapper mapper, ILogger<CampaignService> logger)
     {
-        private readonly ICampaignRepository _campaignRepository;
-        private readonly IMapper _mapper;
+        _campaignRepository = campaignRepository;
+        _mapper = mapper;
+        _logger = logger;
+    }
 
-        public CampaignService(ICampaignRepository campaignRepository, IMapper mapper)
-        {
-            _campaignRepository = campaignRepository;
-            _mapper = mapper;
-        }
+    public async Task<Campaign> CreateCampaignAsync(CampaignPostDTO campaignDto)
+    {
+        _logger.LogInformation("Creating a new campaign.");
 
-        public async Task<Campaign> CreateCampaignAsync(CampaignPostDTO campaignDto)
+        try
         {
             var campaign = _mapper.Map<Campaign>(campaignDto);
             await _campaignRepository.AddAsync(campaign);
+
+            _logger.LogInformation("Campaign created successfully with ID {CampaignId}.", campaign.CampaignID);
             return campaign;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while creating a campaign.");
```

Comment:
logger.LogError("An error occurred while creating a campaign: {ErrorMessage}", ex.Message);

try this format if it works, it is better guidline, even if we read from debugging very easy to aproach